GliaWidgets/Sources/CallVisualizer/CallVisualizer.Coordinator.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+import UIKit
+
+extension CallVisualizer {
+    final class Coordinator {
+        init(
+            viewFactory: ViewFactory,
+            presenter: Presenter
+        ) {
+            self.viewFactory = viewFactory
+            self.presenter = presenter
+        }
+
+        func offerScreenShare(
+            with conf: ScreenShareOfferAlertConfiguration,
+            accepted: @escaping () -> Void,
+            declined: @escaping () -> Void
+        ) {
+            let alert = AlertViewController(
+                kind: .screenShareOffer(conf, accepted: accepted, declined: declined),
+                viewFactory: viewFactory
+            )
+            presenter.getInstance()?.present(alert, animated: true, completion: nil)
+        }
+
+        func presentCallVisualizerViewController() {
+            presenter.getInstance()?.present(
+                CallVisualizer.EngagementViewController(),
+                animated: true
+            )
+        }
+
+        // MARK: - Private
+
+        private let viewFactory: ViewFactory
+        private let presenter: Presenter
+    }
+}
+
+extension CallVisualizer {
+    public struct Presenter {
+        init(presenter: @escaping () -> UIViewController?) {
+            self.getInstance = presenter
+        }
+
+        let getInstance: () -> UIViewController?
+    }
+}
+
+extension CallVisualizer.Presenter {
+    static func topViewController(application: UIKitBased.UIApplication = .live) -> Self {
```

Comment:
I think we should pass `application` parameter explicitly from Glia.Environment. In general we should avoid implicit (default) parameters for dependencies, because they will slip in during unit testing, triggering live API(s). So I wold suggest to remove default `live`.

Reply:
fixed - 19bfbd773c0715dbbd0aa9244fe163c82b6572b9